src/hooks/halving.ts

```diff
@@ -27,6 +27,31 @@ export const useCountdown = (targetDate: Date): [number, number, number, number,
   return [days, hours, minutes, seconds, countdown]
 }
 
+export const useEpochCountdown = (
+  targetEpoch: number,
+): {
+  haveDone: boolean
+  estimatedDate: Date
+} => {
+  const statistics = useStatistics()
+  const currentEpoch = Number(statistics.epochInfo.epochNumber)
+  const epochLength = Number(statistics.epochInfo.epochLength)
+  const epochBlockIndex = Number(statistics.epochInfo.index)
+
+  const { currentEpochEstimatedTime } = useCurrentEpochOverTime(
+    !(currentEpoch === targetEpoch - 1 && epochBlockIndex / epochLength > 0.5),
+  )
+
+  const estimatedTime = currentEpochEstimatedTime + THEORETICAL_EPOCH_TIME * (targetEpoch - currentEpoch - 1)
+  const estimatedDate = useMemo(() => new Date(new Date().getTime() + estimatedTime), [estimatedTime])
+  const haveDone = currentEpoch >= targetEpoch
+
+  return {
+    haveDone,
+    estimatedDate,
+  }
+}
```

Comment:
This hook is used on the hardforkn page, so it shouldn't be defined in halving utils.

Besides, it's a newly defined method, why not reuse the one of halving countdown?

Reply:
@PainterPuppets please check